js/script.js

```diff
@@ -50,19 +50,29 @@ function drawCapacity(map, selectDate){
       sublayers: [{
         // Nested query to only return last record for each reservoir to show current levels in tooltip and speed up query.
         // Converting numeric fields to characters to use formatting in tooltip
-        sql: "SELECT the_geom, the_geom_webmercator, cartodb_id, TO_CHAR(percent_full*100, '90D00') as percent_full, \
-        name, TO_CHAR(reservoir_storage, '9G999G990') as reservoir_storage, storage_capacity, \
-        TO_CHAR(storage_capacity, '9G999G990') as storage_capacity_text, dam_id, date \
-        FROM reservoir_reading_extract \
-        WHERE (dam_id, date) in ( \
-          select dam_id, max(date) \
-          from reservoir_reading_extract \
-          where date <= '"+selectDate+"' \
-          group by dam_id) \
+        sql: "SELECT\
+          the_geom,\
+          the_geom_webmercator,\
+          cartodb_id,\
+          TO_CHAR(percent_full*100, '90D00') as percent_full,\
+          supply_name,\
+          TO_CHAR(supply_storage, '9G999G990') as supply_storage,\
+          storage_capacity,\
+          TO_CHAR(storage_capacity, '9G999G990') as storage_capacity_text,\
```

Comment:
what is the point of the `TO_CHAR`s?

Reply:
the `TO_CHAR`s are there in order to format the text in the tooltip with commas. So because the capacity layer does not need numbers, I just brought them in as text with the formatting already applied.